### PR TITLE
pass-through strategy plugin

### PR DESCRIPTION
**Pull request by jazzyfresh**
This feature would introduce a new strategy plugin for simple pass-through scaling action. That is, whatever the value returned by the APM, this strategy will scale to that value exactly.
```hcl
strategy "pass-through" {}
```

This strategy would be incorporated into a scaling policy like so:
```hcl
scaling "batch_processing_cluster" {
  min = 0
  max = 20

  policy {
    check "queue_size" {
      source = "prometheus"
      query  = "running_jobs + pending_jobs"

      strategy "pass-through" {}
    }

    target "aws-asg" {
      aws_asg_name = "batch_clients"
      node_class = "high_memory"
    }
  }
}
```

### CODE REVIEW

File: plugins/builtin/strategy/none/plugin/plugin.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+// Run satisfies the Run function on the strategy.Strategy interface.
+func (s *StrategyPlugin) Run(eval *sdk.ScalingCheckEvaluation, count int64) (*sdk.ScalingCheckEvaluation, error) {
+
+	// Use only the latest value for now.
+	metric := eval.Metrics[len(eval.Metrics)-1]
+
+
+	// TODO: compare count > metric --> direction
```
Comment (jazzyfresh) [inline plugins/builtin/strategy/none/plugin/plugin.go:69]: remove TODO and maybe replace with a note

File: plugins/builtin/strategy/none/plugin/plugin.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+	return eval, nil
+}
+
+// calculateDirection is used to calculate the direction of scaling that should
+// occur, if any at all. It takes into account the current task group count in
+// order to correctly account for 0 counts.
+//
+// The input factor value is padded by e, such that no action will be taken if
+// factor is within [1-e; 1+e].
```
Comment (jazzyfresh) [inline plugins/builtin/strategy/none/plugin/plugin.go:88]: is this comment still accurate for None strategy?

File: plugins/builtin/strategy/none/plugin/plugin_test.go [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+	"testing"
+	"time"
+
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad-autoscaler/plugins/base"
+	"github.com/hashicorp/nomad-autoscaler/sdk"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStrategyPlugin_SetConfig(t *testing.T) {
```
Comment (jazzyfresh) [inline plugins/builtin/strategy/none/plugin/plugin_test.go:14]: Update tests to match None plugin, not Target-value

File: plugins/plugin.go [hunk 4]
```diff
@@ -23,6 +23,8 @@ const (
 	// InternalAPMPrometheus is the Prometheus APM internal plugin name.
 	InternalAPMPrometheus = "prometheus"
 
+	InternalStrategyNone = "none"
```
Comment (jazzyfresh) [inline plugins/plugin.go:26]: Add description comment like the others

**Comment by jazzyfresh** [comment]
TODO: add documentation to nomadproject.io website in nomad repo

File: plugins/builtin/strategy/none/main.go [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad-autoscaler/plugins"
+	targetvalue "github.com/hashicorp/nomad-autoscaler/plugins/builtin/strategy/target-value/plugin"
+)
+
+func main() {
+	plugins.Serve(factory)
+}
+
+// factory returns a new instance of the TargetValue Strategy plugin.
```
Comment (lgfa29) [inline plugins/builtin/strategy/none/main.go:13]:
```suggestion
// factory returns a new instance of the None Strategy plugin.
```

File: plugins/builtin/strategy/none/main.go [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+func factory(log hclog.Logger) interface{} {
+	return targetvalue.NewTargetValuePlugin(log)
```
Comment (lgfa29) [inline plugins/builtin/strategy/none/main.go:15]: Wrong plugins 😄

File: plugins/builtin/strategy/none/plugin/plugin.go [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+)
+
+var (
+	PluginID = plugins.PluginID{
+		Name:       pluginName,
+		PluginType: sdk.PluginTypeStrategy,
+	}
+
+	PluginConfig = &plugins.InternalPluginConfig{
+		Factory: func(l hclog.Logger) interface{} { return NewTargetValuePlugin(l) },
```
Comment (lgfa29) [inline plugins/builtin/strategy/none/plugin/plugin.go:26]:
```suggestion
		Factory: func(l hclog.Logger) interface{} { return NewNonePlugin(l) },
```

File: plugins/builtin/strategy/none/plugin/plugin.go [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+
+// StrategyPlugin is the None implementation of the strategy.Strategy
+// interface.
+type StrategyPlugin struct {
+	logger hclog.Logger
+}
+
+// NewTargetValuePlugin returns the None implementation of the
+// strategy.Strategy interface.
+func NewTargetValuePlugin(log hclog.Logger) strategy.Strategy {
```
Comment (lgfa29) [inline plugins/builtin/strategy/none/plugin/plugin.go:46]:
```suggestion
// NewNonePlugin returns the None implementation of the
// strategy.Strategy interface.
func NewNonePlugin(log hclog.Logger) strategy.Strategy {
```

File: plugins/builtin/strategy/none/plugin/plugin.go (hunk 1 above)
Comment (lgfa29) [reply plugins/builtin/strategy/none/plugin/plugin.go:69]: Removing is fine 👍

File: plugins/builtin/strategy/none/plugin/plugin.go (hunk 2 above)
Comment (lgfa29) [reply plugins/builtin/strategy/none/plugin/plugin.go:88]: Nope.

**Review by lgfa29 — COMMENTED** [review]
Early review and it's looking good! Just a few places with the wrong plugin name.

**Review by cgbaker — APPROVED** [review]
looks good to me!
why `pass-thru` and not `pass-through`?

**Pull request merged**